kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -30,13 +31,25 @@ public class CheckpointInstance
     /**
      * Placeholder to identify the version that is always the latest on timeline
      */
-    public static final CheckpointInstance MAX_VALUE = new CheckpointInstance(-1);
+    public static final CheckpointInstance MAX_VALUE = new CheckpointInstance(Long.MAX_VALUE);
 
     public final long version;
     public final Optional<Integer> numParts;
 
     public CheckpointInstance(Path path) {
-        this(FileNames.getFileVersion(path));
+        String[] pathParts = path.getName().split("\\.");
+
+        if (pathParts.length == 3 && pathParts[1].equals("checkpoint") &&
```

Comment:
For each of these if branches, i'd comment an example to give a visual explanation of what is happening.
e.g. `000123.checkpoint.parquet`.

Also, for `00000000000000000010.checkpoint.0000000002.0000000003.parquet` do we not capture the `002`? We only save the version and the numParts?